client/httpClient.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+)
+
+var client *http.Client
+
+// Permission encapsulates the tls configurations.
+type Permission struct {
+	CertFile string
+	KeyFile  string
+	CaCert   string
+}
+
+// GetCaCert returns the CertFile if CaCert is empty.
+func (perm *Permission) GetCaCert() string {
+	if perm.CaCert == "" {
+		return perm.CertFile
+	}
+	return perm.CaCert
+}
+
+// SendRequest encapsulates the entire process of initializing the client
+// and sending a request, returning the byte body of the response.
+func SendRequest(permission Permission, request *http.Request) (*[]byte, error) {
```

Comment:
It is uncommon to return a pointer to a slice or an array as these are by themselves reference types so returning them by value is fine since you can use `nil` in this case as well.

Reply:
thanks, good to know. this is valuable information